### PR TITLE
[JSON-API] Perf gatling MultiUserQueryScenario

### PR DESCRIPTION
**Pull request by akshayshirahatti-da**
The multi-user scenario used for connection pool testing.

#### Oracle docker vm properties
```
vCPU: 8
mem: 8GB
swap: 2GB
```

#### Miscellaneous Notes

 - Fetch by key run Mode was slower then simple id Queries, need to re-evaluate if its specific to the daml design. Also needed tuning akka timeouts.

https://docs.google.com/spreadsheets/d/1HNqzOlMJX0X963DSGDZAYmk3JbWJHwvCbC4TtkUP9iQ/edit?usp=sharing
----
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/http-json-perf/daml/LargeAcs.daml [hunk 1]
```diff
@@ -114,3 +114,18 @@ template NotIou
...
+    owner : Party
+    currency : Text
+    amount : Decimal
+    observers : [Party]
+  where
+    signatory issuer, owner
+    observer observers
+
+    key (issuer, id): (Party, Int)
+    maintainer key._1
```
Comment (victormueller-da) [inline ledger-service/http-json-perf/daml/LargeAcs.daml:131]: newline missing at the end of the file, can you add it?

**Review by cocreature — APPROVED** [review]
Thanks!

File: ledger-service/http-json-perf/src/main/scala/com/daml/http/perf/scenario/MultiUserQueryScenario.scala [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  final val currencies = List("USD", "GBP", "EUR", "CHF", "AUD")
+
+  def randomCurrency(): String = {
+    this.synchronized { rng.shuffle(currencies).head }
+  }
+}
+
+object MultiUserQueryScenario {
+  sealed trait RunMode { def name: String }
+  case object PopulateCache extends RunMode { val name = "populateCache" }
```
[the remark this answers is not in the document]
Comment (akshayshirahatti-da) [reply ledger-service/http-json-perf/src/main/scala/com/daml/http/perf/scenario/MultiUserQueryScenario.scala:24]: I think a regular scala binary can work as well , but I guess we can use this as a write side test as well if we want to test that via the JSON API.

File: ledger-service/http-json-perf/daml/LargeAcs.daml (hunk 1 above)
Comment (akshayshirahatti-da) [reply ledger-service/http-json-perf/daml/LargeAcs.daml:131]: I guess my use of mergify wasn't the best here , I should have used that tag after the initial review. I will make this change in my followup PR's

**Comment by akshayshirahatti-da** [comment]
work done as part of [#10155](https://github.com/digital-asset/daml/issues/10155)

**Pull request merged**